test/val/val_decoration_test.cpp

```diff
@@ -548,4 +548,179 @@ OpFunctionEnd
+// layout(binding = 1) uniform sampler2D s2d[][2];
+// layout(location = 0) in nonuniformEXT int i;
+// void main()
+// {
+//     vec4 v = texture(s2d[i][i], vec2(0.3));
+// }
+TEST_F(ValidateDecorations, RuntimeArrayOfArraysOfDescriptorSetsIsDisallowed) {
+  const spv_target_env env = SPV_ENV_VULKAN_1_0;
+  std::string spirv = R"(
+               OpCapability Shader
+               OpCapability ShaderNonUniformEXT
+               OpCapability RuntimeDescriptorArrayEXT
+               OpCapability SampledImageArrayNonUniformIndexingEXT
+               OpExtension "SPV_EXT_descriptor_indexing"
+          %1 = OpExtInstImport "GLSL.std.450"
+               OpMemoryModel Logical GLSL450
+               OpEntryPoint Vertex %main "main" %i
+               OpSource GLSL 440
+               OpSourceExtension "GL_EXT_nonuniform_qualifier"
+               OpName %main "main"
+               OpName %v "v"
+               OpName %s2d "s2d"
+               OpName %i "i"
+               OpDecorate %s2d DescriptorSet 0
+               OpDecorate %s2d Binding 1
+               OpDecorate %i Location 0
+               OpDecorate %i NonUniformEXT
+               OpDecorate %21 NonUniformEXT
+               OpDecorate %22 NonUniformEXT
+               OpDecorate %25 NonUniformEXT
+       %void = OpTypeVoid
+          %3 = OpTypeFunction %void
+      %float = OpTypeFloat 32
+    %v4float = OpTypeVector %float 4
+%_ptr_Function_v4float = OpTypePointer Function %v4float
+         %10 = OpTypeImage %float 2D 0 0 0 1 Unknown
+         %11 = OpTypeSampledImage %10
+       %uint = OpTypeInt 32 0
+     %uint_2 = OpConstant %uint 2
+%_arr_11_uint_2 = OpTypeArray %11 %uint_2
+%_runtimearr__arr_11_uint_2 = OpTypeRuntimeArray %_arr_11_uint_2
+%_ptr_UniformConstant__runtimearr__arr_11_uint_2 = OpTypePointer UniformConstant %_runtimearr__arr_11_uint_2
+        %s2d = OpVariable %_ptr_UniformConstant__runtimearr__arr_11_uint_2 UniformConstant
+        %int = OpTypeInt 32 1
+%_ptr_Input_int = OpTypePointer Input %int
+          %i = OpVariable %_ptr_Input_int Input
+%_ptr_UniformConstant_11 = OpTypePointer UniformConstant %11
+    %v2float = OpTypeVector %float 2
+%float_0_300000012 = OpConstant %float 0.300000012
+         %28 = OpConstantComposite %v2float %float_0_300000012 %float_0_300000012
+    %float_0 = OpConstant %float 0
+       %main = OpFunction %void None %3
+          %5 = OpLabel
+          %v = OpVariable %_ptr_Function_v4float Function
+         %21 = OpLoad %int %i
+         %22 = OpLoad %int %i
+         %24 = OpAccessChain %_ptr_UniformConstant_11 %s2d %21 %22
+         %25 = OpLoad %11 %24
+         %30 = OpImageSampleExplicitLod %v4float %25 %28 Lod %float_0
+               OpStore %v %30
+               OpReturn
+               OpFunctionEnd
+)";
+  CompileSuccessfully(spirv, env);
+
+  EXPECT_EQ(SPV_ERROR_INVALID_ID, ValidateAndRetrieveValidationState(env));
+  EXPECT_THAT(getDiagnosticString(),
+              HasSubstr("Array of arrays is not allowed"));
```

Comment:
I think you forgot to update this error message.

Reply:
Sigh, yes I did. Thanks.